mm2src/coins/eth.rs

```diff
@@ -5503,16 +5676,51 @@ pub async fn eth_coin_from_conf_and_request(
 
     let mut web3_instances = vec![];
     let event_handlers = rpc_event_handlers_for_eth_transport(ctx, ticker.to_string());
-    for node in nodes.iter() {
-        let transport = Web3Transport::new_http(vec![node.clone()], event_handlers.clone());
+    for url in urls.iter() {
+        let uri: Uri = try_s!(url.parse());
+
+        let transport = match uri.scheme_str() {
+            Some("ws") | Some("wss") => {
```

Comment:
We don't have to support new schemes or functionalities for legacy APIs, it's good that you added websocket support here but you can keep that in mind in the future to reduce the workload. If any legacy API has a v2 alternative we should add new functionalities to the v2 only as long as nothing breaks for the legacy ones.